sdcflows/cli/main.py

```diff
@@ -0,0 +1,182 @@
+    # straightforward way to communicate with the child process is via the filesystem.
+    # The config file name needs to be unique, otherwise multiple sdcflows instances
+    # will create write conflicts.
+    config_file = mktemp(
+        dir=config.execution.work_dir, prefix=".sdcflows.", suffix=".toml"
+    )
+    config.to_filename(config_file)
+    config.file_path = config_file
+    exitcode = 0
+
+    if config.workflow.analysis_level != ["participant"]:
+        raise ValueError("Analysis level can only be 'participant'")
+
+    if config.execution.dry_run:  # --dry-run: pretty print results
+        from niworkflows.utils.bids import collect_participants
+        from sdcflows.utils.wrangler import find_estimators
+
+        subjects = collect_participants(
+            config.execution.layout,
+            config.execution.participant_label,
+        )
+        estimators_record = {}
+        for subject in subjects:
+            estimators_record[subject] = find_estimators(
+                layout=config.execution.layout,
+                subject=subject,
+                fmapless=config.workflow.fmapless,
+                logger=config.loggers.cli,
+            )
+
+        print(f"Estimation for <{config.execution.bids_dir}> complete. Found:")
+        for subject, estimators in estimators_record.items():
+            print(f"\tsub-{subject}")
+            if not estimators:
+                print("\t\tNo estimators found")
+                continue
+            for estimator in estimators:
+                print(f"\t\t{estimator}")
+                for fl in estimator.sources:
+                    fl_relpath = fl.path.relative_to(config.execution.bids_dir / f"sub-{subject}")
+                    pe_dir = fl.metadata.get("PhaseEncodingDirection")
+                    print(f"\t\t\t{pe_dir}\t{fl_relpath}")
+        sys.exit(exitcode)
+
+    # Initialize process pool if multiprocessing
+    _pool = None
+    if config.nipype.plugin in ("MultiProc", "LegacyMultiProc"):
+        from contextlib import suppress
+        import multiprocessing as mp
+        import multiprocessing.forkserver
+        from concurrent.futures import ProcessPoolExecutor
+
+        os.environ["OMP_NUM_THREADS"] = "1"
+
+        with suppress(RuntimeError):
+            mp.set_start_method("fork")
+        gc.collect()
+
+        _pool = ProcessPoolExecutor(
+            max_workers=config.nipype.nprocs,
+            initializer=config._process_initializer,
+            initargs=(config.file_path,),
+        )
+
+    if not config.execution.notrack:
+        from sdcflows.utils.telemetry import setup_migas
+
+        setup_migas()
+
+    # CRITICAL Call build_workflow(config_file, retval) in a subprocess.
+    # Because Python on Linux does not ever free virtual memory (VM), running the
+    # workflow construction jailed within a process preempts excessive VM buildup.
+    from multiprocessing import Manager, Process
+
+    with Manager() as mgr:
```

Comment:
Okay, I was removing this context to build the workflow in the main thread, and it could be interesting to keep it this way, as the workflow is created while OMP_NUM_THREADS is set to 1. This saves great numpy buildup. 

We could still move it to the main process, but then it would make sense to make sure numpy is imported with OMP_NUM_THREADS = 1 (and that is then overridden in  subprocesses).